scripts/evalCrawler.R

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env Rscript
+
+library(stringr)
+library(readr)
+library(purrr)
+library(dplyr)
+library(pbapply)
+
+#Sys.setlocale('LC_ALL','C')
+regex <- list()
+
+regex[[1]]  <- c( "(?<!([\"a-zA-Z0-9_#]))(eval\\()" , "4" , "eval")
+regex[[2]]  <- c( "(?<!([\"a-zA-Z0-9_#]))(evalq\\()" , "5" , "evalq")
+regex[[3]]  <-  c("(?<!([\"a-zA-Z0-9_#]))(eval\\.parent\\()" , "11" , "eval.parent")
+regex[[4]]  <- c("(?<!([\"a-zA-Z0-9_#]))(local\\()" , "5" , "local")
+
+fileCrawlOne <- function(file, regex, pkgName) {
+  lines <- readLines(file)
+  lineCount <- length(lines)
+  lines <- gsub("(^(([\\s\\t]*)?#)).*", "\n", lines , perl = TRUE);
+  lineNs <- lineNumbers(lines , regex)
+  
+  str <- paste0(lines, collapse="\n")
+  
+  df <- returnEvalsFromString(lineNs , str , pkgName , file , lineCount , regex)
+  return(df)
+}
+
+evalCrawlOne <- function(dir , libAddr , regex) {
+  #print("1")
+  print(dir)
+  
+  pkgName <- dir
+  pkgAddr <- paste(libAddr , pkgName , sep="/")
+  fileLists <- list.files(path = pkgAddr , recursive = TRUE , pattern = "\\.((R)|(Rd)|(Rmd)|(rmd)|(r)|(rd))$", full.names = TRUE)
+  #fileLists <- list.files(path = pkgAddr , recursive = TRUE , full.names = TRUE)
+  df<- map_dfr(fileLists, ~ fileCrawlOne(., regex , pkgName))
+  return(df)
+}
+
+evalCrawl <- function( regex , libAddr) {
+  dirLists <- list.dirs(path = libAddr , full.names = FALSE , recursive = FALSE)
+  print(length(dirLists))
+  dfs <- pblapply(dirLists, evalCrawlOne, libAddr=libAddr, regex = regex , cl=parallel::detectCores())
+  
+  df <- bind_rows(dfs)
+  return(df)
+}
+
+main <- function(libAddr){
+  dfs <- lapply(regex , evalCrawl , libAddr = libAddr)
+  df <- bind_rows(dfs)
+  write_csv(df, "evals.csv")
+}
+
+lineNumbers <- function(lst , regex)
```

Comment:
What is this suppose to do?

Reply:
This lists the line numbers of `regex` matching strings in the vector `lst`. 
If there are multiple occurences in a single vetor element, the index (or the line number) is repeated that many times.